src/app/pages/upload.ts

```diff
@@ -0,0 +1,71 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import xlsx from 'xlsx';
+import multer from 'multer';
+import { Request, Response } from 'express';
```

Comment:
I think the correct types would be `import type { NextApiRequest, NextApiResponse } from 'next'` which are a little different from Express's `Request`/`Response` types. However, this is a Next.js Page router convention, whereas the App router convention would be like:
```
import { NextResponse } from 'next/server'
 
export async function POST(request: Request) {
  ...
  return NextResponse.json({ data })
}
```